crates/biome_js_analyze/src/analyzers/nursery/no_evolving_any.rs

```diff
@@ -0,0 +1,121 @@
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_js_syntax::{JsFileSource, JsVariableDeclaration, JsVariableDeclarator};
+
+declare_rule! {
+    /// Disallow variables from evolving into `any` type through reassignments.
+    ///
+    /// In TypeScript, variables without explicit type annotations can evolve their types based on subsequent assignments.
+    /// This behavior can inadvertently lead to variables with an `any` type, weakening type safety.
```

Comment:
IMHO, the word accidentally is used more frequently than inadvertently.
```suggestion
    /// This behavior can accidentally lead to variables with an `any` type, weakening type safety.
```